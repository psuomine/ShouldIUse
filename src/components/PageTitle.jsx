import React from 'react'
import PropTypes from 'prop-types'

const PageTitle = ({ title }) => <h1>{title}</h1>

PageTitle.propTypes = {
  title: PropTypes.string.isRequired,
}

export default PageTitle
