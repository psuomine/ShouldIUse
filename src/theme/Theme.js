const Theme = {
  background: '#F8F7F9',
  primary: '#651EFF',
  disabled: 'rgba(0,0,0,.12)',
  successBackground: 'rgba(23,195,123, .12)',
  success: 'rgba(23,195,123)',
  typography: {
    text: '#141033',
    textSecondary: '#9171D6',
    white: '#fff',
    muted: 'rgba(0,0,0,.54)',
    primary: '#651EFF',
    error: '#E74C3C',
  },
}

export default Theme
