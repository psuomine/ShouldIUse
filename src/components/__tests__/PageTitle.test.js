import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import Theme from 'theme/theme'
import PageTitle from '../PageTitle'

afterEach(cleanup)

const renderComponent = title =>
  render(
    <ThemeProvider theme={Theme}>
      <PageTitle title={title} />
    </ThemeProvider>
  )

describe('<PageTitle />', () => {
  it('renders h1 element', () => {
    const { container } = renderComponent('Headline')
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders title based on props', () => {
    const title = 'Long title'
    const { getByText } = renderComponent(title)
    expect(getByText(title).textContent).toEqual(title)
  })
})
