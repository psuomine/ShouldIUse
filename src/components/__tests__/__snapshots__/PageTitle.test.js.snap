// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> renders h1 element 1`] = `
<h1
  class="sc-bdVaJa eAkhWA"
>
  Headline
</h1>
`;
